tests/unit_tests/client/api/i2p_control/parser.cc

```diff
@@ -0,0 +1,802 @@
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF    //
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.               //
+ *                                                                                            //
+ */
+
+#define BOOST_TEST_DYN_LINK
+
+#include <boost/test/unit_test.hpp>
+
+#include <array>
+#include <limits>
+
+#include "client/api/i2p_control/packet.h"
+#include "client/api/i2p_control/parser.h"
+#include "client/util/json.h"
+
+namespace client = kovri::client;
+namespace core = kovri::core;
+
+struct I2PControlSessionFixture
+{
+  // For convenience
+  typedef client::I2PControlResponse Response;
+  typedef client::I2PControlRequest Request;
+  typedef Response::Method Method;
+  typedef Response::Keys Keys;
+  typedef Response::Params Params;
+  typedef Response::I2PControl I2PControl;
+  typedef Response::RouterInfo RouterInfo;
+  typedef Response::RouterManager RouterManager;
+  typedef Response::NetworkSetting NetworkSetting;
+  typedef Response::NetStatus NetStatus;
+
+  I2PControlSessionFixture()
+  {
+    // Initialize json
+    m_Json["3886212441"]["bytes"] = client::JsonObject(0);
+    m_Json["3886212441"]["layout"] =
+        client::JsonObject("me-->3886212441:nrkY-->");
+  }
+
+  // Used values
+  const std::size_t m_Id{123};
+  const std::size_t m_Api{456};
+  const std::string m_EchoMessage{"echo message"};
+  const std::string m_Token{"some_token"};
+  const std::string m_Password{"some_pass"};
+  const std::string m_Version{"2.0"};
+  const std::string m_Address{"172.18.0.10"};
+  const std::string m_Port{"15150"};
+  const std::string m_Status{"some status"};
+  const std::size_t m_Uptime{123456};
+  const std::string m_KovriVersion{"some version"};
+  const double m_BWIn1S{1.1};
+  const double m_BWIn15S{15.15};
+  const double m_BWOut1S{2.2};
+  const double m_BWOut15S{25.25};
+  Response::NetStatus m_NetStatus{NetStatus::Firewalled};
+  const std::size_t m_Participants{5};
+  const std::size_t m_ActivePeers{10};
+  const std::size_t m_FastPeers{7};
+  const std::size_t m_HighCapPeers{3};
+  const std::size_t m_KnownPeers{50};
+  const std::string m_DataPath{"/path/to/data/dir"};
+  const std::size_t m_Floodfills{20};
+  const std::size_t m_LeaseSets{30};
+  const double m_TunnelsCreationSuccessRate{0.83};
+  const std::string m_StatusUpdate{"update status"};
+  const std::string m_NTCPPort{"25250"};
+  const std::string m_NTCPHostName{"ntcp hostname"};
+  const std::string m_NTCPAutoIP{"ntcp auto ip"};
+  const std::string m_SSUPort{"25251"};
+  const std::string m_SSUHostName{"ssu hostname"};
+  const std::string m_SSUAutoIP{"ssu auto ip"};
+  const std::string m_UPNP{"upnp"};
+  const std::string m_BWShare{"BW share"};
+  const std::string m_BWIn{"10.10"};
+  const std::string m_BWOut{"20.20"};
+  const std::string m_LaptopMode{"laptop mode"};
+  client::JsonObject m_Json;
```

Comment:
For this amount of copypasta'd types, why not organize these various types into structures instead (i.e. RI participants are only for an RI Response message type)?